src/ros_filter.cpp

```diff
@@ -660,23 +660,26 @@ namespace RobotLocalization
 
   template<typename T>
   void RosFilter<T>::trustedSensorCallback(const std_msgs::Bool::ConstPtr &msg,
-                                           const std::string &topicName)
+                                           const std::vector<std::string> &topicNames)
   {
-    // RF_VERBOSE provides this info in the debug file inline with the received and
-    //  and processed data, so is highly useful
-    RF_VERBOSE("Received trusted sensor data for topic " << topicName <<
-      " with trusted " << ((msg->data) ? "true\n" : "false\n"));
-    // If this information is to be provided via a ROS stream, do so from the provider
-    // Pass it in/save it all
-    if(sourceData_.find(topicName) == sourceData_.end())
-    {
-      RF_VERBOSE("Adding trusted sensor source data for topic " << topicName << "\n");
-      // Create it
-      sourceData_.emplace(topicName, SourceData());
+    for ( auto topicName : topicNames)
+    {
+      // RF_VERBOSE provides this info in the debug file inline with the received and
+      //  and processed data, so is highly useful
+      RF_VERBOSE("Received trusted sensor data for topic " << topicName <<
+        " with trusted " << ((msg->data) ? "true\n" : "false\n"));
+      // If this information is to be provided via a ROS stream, do so from the provider
+      // Pass it in/save it all
+      if(sourceData_.find(topicName) == sourceData_.end())
+      {
+        RF_VERBOSE("Adding trusted sensor source data for topic " << topicName << "\n");
+        // Create it
+        sourceData_.emplace(topicName, SourceData());
+      }
+      // Save the data
+      sourceData_[topicName].trusted_ = msg->data;
+      sourceData_[topicName].last_trusted_s_ = ros::Time::now().toSec();
```

Comment:
Simply put that within the for loop so that the trusted data could be applied to the respective data being processed.
No other changes were made to that.